### ISSUE TITLE
API taken down

### ISSUE THREAD

**Issue by comistet** [bobbymond/FantasyPremierLeagueAPI#3]
It looks like the fantasy stats API (e.g. https://fantasy.premierleague.com/web/api/elements/222/) has been taken down. Do you know how else we can get this data?

**Comment by bobbymond** [comment]
Looks like there is something here: https://fantasy.premierleague.com/drf/element-summary/
Haven't dug in yet to check if that's the same thing or not.  Looks like the whole site has been redesigned so might need to fix some of the other site scraping stuff.

**Comment by lazytarget** [comment]
I've researched a bit and the data can easily be accessed by simple GET requests.

Base url: https://fantasy.premierleague.com/drf/

Relative links:
/bootstrap (more data if authenticated)
/bootstrap-static
/bootstrap-dynamic

/events
/elements
/element-types
/fixtures
/teams
/region
/transfers (requires auth)
/entry/{entryId}
/entries (doesn't work in pre-season?)
/my-team/{teamId} (requires auth)
/leagues-entered/{teamId} (requires auth)
/leagues-classic/{leagueId} (must be a member)
/leagues-classic-standings/{leagueId}
/leagues-h2h/{leagueId} (must be a member)
/leagues-h2h-standings/{leagueId}

I'll see if I have the time to try and implement these in a pull-request.

**Comment by lazytarget** [comment]
Don't know if there are any API's for updating the squad or making transfers.

Also not entirely sure whether if the authentication is handled by cookie or through request credentials.

**Comment by lazytarget** [comment]
I have begun with some updates on [my fork](https://github.com/LazyTarget/FantasyPremierLeagueAPI).

**Comment by mattdunndc** [comment]
Any idea what the columns ep_this, ep_next or ea_index mean?

**Comment by gpoudel** [comment]
"Estimated points this Game Week"  and "Estimated Points Next Game Week". They could be working on some predictive modeling idea but this could be very simple formula.

**Comment by tommed** [comment]
Can I suggest doing a `HEAD` request rather than a `GET` will rapidly improve performance. When it returns a 200 the player exists, when it returns a 404, they don't. Then you can request the GETs.

**Comment by Qd7** [comment]
if any is looking to get data in excel,

using Lazytarget post, i found this
https://fantasy.premierleague.com/drf/entry/{teamId}/history

you can use excel power query to parse the json and link it to excel as a table

**Comment by brood915** [comment]
Hm how did you guys deal with the CORS problem? I cant access any data

**Comment by andrew3stedall** [comment]
Would you be able to help with authentication over php?

**Comment by tommed** [comment]
@brood915, you either request these at the server (so CORs doesn't apply), and/or create you're own proxy with CORs enabled. Either way, you'll need a server component to delegate the calls, you can't just develop a front-end and plug into these API calls.

**Comment by tanushshukla** [comment]
@LazyTarget ,
I checked your fork as well. It's not working.
https://github.com/LazyTarget/FantasyPremierLeagueAPI

Is it normal? Didn't debug much but only one cookie is getting set.

**Comment by al3krab1** [comment]
/transfers (requires auth) !!
how can i get auth in my website?

**Comment by Heri88** [comment]
Anyone able to get auth api?

**Comment by PierreBoislard3** [comment]
Anyone found how to get auth on external website?

**Comment by chris-mann-uk** [comment]
Has anybody taken a look at this recently?

The API appears to have changed again, leaving all links previously mentioned null and void.

**Comment by ap752-unikent** [comment]
> Has anybody taken a look at this recently?
> 
> The API appears to have changed again, leaving all links previously mentioned null and void.

Yeah, I've noticed this as well. I'm hoping this is just because the season hasn't started yet. 
If the API has changed however, how did you guys find the new one last time?

**Comment by ap752-unikent** [comment]
nvm, I think I found it

https://fantasy.premierleague.com/api/bootstrap-static/


For future reference, if you look at the network tab on chrome and record the traffic coming in when you load the page, then you can view all the pages being called. The api calls will be here

**Comment by lastmaj** [comment]
> nvm, I think I found it
> 
> https://fantasy.premierleague.com/api/bootstrap-static/
> 
> For future reference, if you look at the network tab on chrome and record the traffic coming in when you load the page, then you can view all the pages being called. The api calls will be here

I think it loads just the one. 
The only other endpoint I could have found was : https://fantasy.premierleague.com/api/fixtures/. Please share if you can find others as well.

**Comment by ScottBruton** [comment]
Hey guys, I carry on receiving “Authentication credentials were not provided” , and then I can’t access any data from leagues.. 
can anyone help?

**Comment by ap752-unikent** [comment]
> nvm, I think I found it
> https://fantasy.premierleague.com/api/bootstrap-static/
> For future reference, if you look at the network tab on chrome and record the traffic coming in when you load the page, then you can view all the pages being called. The api calls will be here
> 
> I think it loads just the one.
> The only other endpoint I could have found was : https://fantasy.premierleague.com/api/fixtures. Please share if you can find others as well.

You can also find each player's individual attributes by sending a request to "https://fantasy.premierleague.com/api/element-summary/{player-id}/"

Each players id can be found via the api call to boostrap static

**Comment by ap752-unikent** [comment]
> Hey guys, I carry on receiving “Authentication credentials were not provided” , and then I can’t access any data from leagues..
> can anyone help?

That's odd, still working for me. Just to make sure, you're sending a GET request not a POST right?

**Comment by Qd7** [comment]
Hi all,

any chance anyone can help us with step-by-step import these json links into Excel?

Regards
________________________________

**Comment by lastmaj** [comment]
> > Hey guys, I carry on receiving “Authentication credentials were not provided” , and then I can’t access any data from leagues..
> > can anyone help?
> 
> That's odd, still working for me. Just to make sure, you're sending a GET request not a POST right?

Yes, if a an authenticated user browses his league standings, the league standings API endpoint returns data but only for a minute or so, but generally it requires authentication. 
Basically you need to send a post req with authentication details to a certain URL and then capture the cookies from the response and use them for all subsequent queries : https://medium.com/@bram.vanherle1/fantasy-premier-league-api-authentication-guide-2f7aeb2382e4
